libsolidity/ast/AST.h

```diff
@@ -2596,6 +2596,35 @@ class TypeClassName: public ASTNode
 	std::variant<Token, ASTPointer<IdentifierPath>> m_name;
 };
 
+class BuiltinDefinition: public Expression
```

Comment:
Actually, I'm not entirely sure about this name - is this really a "definition" :-)? We could just have plainly called it ``Builtin`` or ``BuiltinExpression`` - but let's not hold up the PR on this.

Reply:
Yeah, good point. `__builtin()` is not the whole type definition, just an expression. Your names are better. I'll rename it.

Reply:
Renamed to `Builtin`.